packages/flutter_tools/lib/src/runner/flutter_command_runner.dart

```diff
@@ -455,7 +458,9 @@ class FlutterCommandRunner extends CommandRunner<void> {
     // * strip '_sim_' since there are no host simulator builds.
     // * replace the target platform with host.
     final String basename = fs.path.basename(engineBuildPath);
-    final String hostBasename = 'host_' + basename.replaceFirst('_sim_', '_').substring(basename.indexOf('_') + 1);
+    String tmpBasename = basename.replaceFirst('_sim_', '_').substring(basename.indexOf('_') + 1);
+    tmpBasename = tmpBasename.replaceFirst(RegExp(r'_arm64$'), '').replaceFirst(RegExp(r'_arm$'), '');
```

Comment:
Can you also strip x86 and x64? Also update the comment above

Reply:
Will have a check towards the x86&x64 logic before merging this pr.

Reply:
@liamappelbe 
Fixed, please review it.